packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderContent/ConditionBuilderContent.tsx

```diff
@@ -46,6 +53,10 @@ const ConditionBuilderContent = ({
 }: ConditionBuilderContentProps) => {
   const { rootState, setRootState, variant, actionState } =
     useContext<ConditionBuilderContextProps>(ConditionBuilderContext);
+
+  const initialConditionState = useRef(
```

Comment:
i don't think this is still quite right. if `initialState` is being passed as a prop then there's no need to store the value again anywhere. not in state or as a ref. if this is simply needed to establish the base state in `rootState` then i think it just needs to be set once in the effect on line 100.

Reply:
I got your point. This is to fix this issue. Consider we pass an  initial state prop and it will populate that in the condition builder. And now if you remove all conditions till no condition , we show _Add Condition_ button., and again when i click on _add condition_ ,this time  the builder should start from an empty state , rather than from the initial state. so we cannot use initial state prop directly in all scenarios, and since prop is immutable I had to either use a state/ref.
Current behaviour: 

https://github.com/user-attachments/assets/551b98d2-66f2-4527-b95d-bc01bf1d7857

Updated behaviour:

https://github.com/user-attachments/assets/7c968708-8bd1-4105-8dfb-244eb2bea95e

Reply:
ok i see. makes sense